src/agent0/core/hyperdrive/interactive/i_hyperdrive.py

```diff
@@ -205,13 +205,12 @@ def _init_agent(
 
     def _set_max_approval(self, agent: HyperdriveAgent):
         # Establish max approval for the hyperdrive contract
-        asyncio.run(
-            set_max_approval(
-                [agent],
-                self.interface.web3,
-                self.interface.base_token_contract,
-                str(self.interface.hyperdrive_contract.address),
-            )
+        set_max_approval(
```

Comment:
This is confusing me. On main, there are 2 spots where this function is being called: one in `get_agent_accounts.py`, where we pass in a list of agents, and one in `i_hyperdrive.py`, where we pass in a single agent. Making this function to only take a single agent makes me wonder what happened to the other call in this PR

Reply:
It looks like this was redefined in `setup_experiments` as a private function. So this PR is making a change where we were reusing one function to now using two separate functions.